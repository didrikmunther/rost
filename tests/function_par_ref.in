let b = 123;

fn print_and_change(k: pointer<pointer<int>>) {
	printf("%i\n", **k);
	**k = 5;
}

fn get_b_location() -> pointer<int> {
	return &b;
}

fn main() {
	let i = &b;
	print_and_change(&i);
	printf("%i\n", b);

	let b_location = get_b_location();
	*b_location = 567;
	printf("%i\n", b);

	*get_b_location() = 777;
	printf("%i\n", b);
}