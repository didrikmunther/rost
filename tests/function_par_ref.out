123
5
567
777